### PR TITLE
Support ddtrace 1.x

### PR DESCRIPTION
**Pull request by gedorinku**
## Why

ddtrace 0.x doesn't support Ruby 3.2.
https://github.com/DataDog/dd-trace-rb/blob/9fd04441f91f4668c6d31dcff2af2115b38318c0/lib/ddtrace/version.rb#L13-L23

## What

Support ddtrace 1.x.

### CODE REVIEW

File: lib/opencensus/trace/exporters/datadog/converter.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
-require 'ddtrace/ext/distributed'
-require 'ddtrace/ext/priority'
-require 'ddtrace/ext/errors'
+require 'ddtrace/version'
```
Comment (qnighy) [inline lib/opencensus/trace/exporters/datadog/converter.rb:1]: Maybe it is now unnecessary as you have removed version-based branches.

File: lib/opencensus/trace/exporters/datadog/converter.rb [hunk 2]
```diff
@@ -13,7 +12,7 @@ class Converter
           DATADOG_SPAN_TYPE_KEY = 'span.type'.freeze
           DATADOG_SERVICE_NAME_KEY = 'service.name'.freeze
           DATADOG_RESOURCE_NAME_KEY = 'resource.name'.freeze
-          DATADOG_SAMPLING_PRIORITY_KEY = ::Datadog::Ext::DistributedTracing::SAMPLING_PRIORITY_KEY
+          DATADOG_SAMPLING_PRIORITY_KEY = ::Datadog::Tracing::Metadata::Ext::Distributed::TAG_SAMPLING_PRIORITY
```
Comment (gedorinku) [inline lib/opencensus/trace/exporters/datadog/converter.rb:15]: 0.x: https://github.com/DataDog/dd-trace-rb/blob/9fd04441f91f4668c6d31dcff2af2115b38318c0/lib/ddtrace/ext/distributed.rb#L10
1.x: https://github.com/DataDog/dd-trace-rb/blob/4a1050cd7dd900eaabe8f40f03e1db1b905f8c35/lib/datadog/tracing/metadata/ext.rb#L53

File: lib/opencensus/trace/exporters/datadog/converter.rb [hunk 3]
```diff
@@ -57,7 +56,7 @@ def convert_span(span)
               duration: ((span.end_time.to_f - span.start_time.to_f) * 1e9).to_i,
             }
 
-            dd_span[:metrics][DATADOG_SAMPLING_PRIORITY_KEY] = ::Datadog::Ext::Priority::AUTO_KEEP
+            dd_span[:metrics][DATADOG_SAMPLING_PRIORITY_KEY] = ::Datadog::Tracing::Sampling::Ext::Priority::AUTO_KEEP
```
Comment (gedorinku) [inline lib/opencensus/trace/exporters/datadog/converter.rb:59]: 0.x: https://github.com/DataDog/dd-trace-rb/blob/9fd04441f91f4668c6d31dcff2af2115b38318c0/lib/ddtrace/ext/priority.rb#L13
1.x: https://github.com/DataDog/dd-trace-rb/blob/4a1050cd7dd900eaabe8f40f03e1db1b905f8c35/lib/datadog/tracing/sampling/ext.rb#L20

File: lib/opencensus/trace/exporters/datadog/converter.rb [hunk 4]
```diff
@@ -92,7 +91,7 @@ def convert_status(dd_span, status)
             status_key = STATUS_DESCRIPTION_KEY
             return if status.nil?
             if status.code != 0 then
-              status_key = ::Datadog::Ext::Errors::MSG
+              status_key = ::Datadog::Tracing::Metadata::Ext::Errors::TAG_MSG
```
Comment (gedorinku) [inline lib/opencensus/trace/exporters/datadog/converter.rb:94]: 0.x: https://github.com/DataDog/dd-trace-rb/blob/9fd04441f91f4668c6d31dcff2af2115b38318c0/lib/ddtrace/ext/errors.rb#L6
1.x: https://github.com/DataDog/dd-trace-rb/blob/4a1050cd7dd900eaabe8f40f03e1db1b905f8c35/lib/datadog/tracing/metadata/ext.rb#L63

**Comment by qnighy** [comment]
https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md#from-0x-to-10

**Pull request merged**